xtext/org.icyphy.linguafranca.tests/src/org/icyphy/tests/runtime/TestBase.xtend

```diff
@@ -0,0 +1,395 @@
+                    binaryName = nameOnly + ".exe"
+                }
+
+                val fullPath = binPath.resolve(binaryName)
+                if (Files.exists(fullPath)) {
+                    // Running the command as .\binary.exe does not work on Windows for
+                    // some reason... Thus we simply pass the full path here, which
+                    // should work across all platforms
+                    pb = new ProcessBuilder(fullPath.toString)
+                    pb.directory(binPath.toFile)
+                } else {
+                    test.issues.append(
+                        fullPath + ": No such file or directory." + NEW_LINE)
+                    test.result = Result.NO_EXEC_FAIL
+                }
+            }
+            case Python: {
+                val srcGen = test.fileConfig.getSrcGenPath
+                val fullPath = srcGen.resolve(nameOnly + ".py")
+                if (Files.exists(fullPath)) {
+                    pb = new ProcessBuilder("python3", fullPath.toFile.name)
+                    pb.directory(srcGen.toFile)
+                } else {
+                    test.result = Result.NO_EXEC_FAIL
+                    if (pb !== null) {
+                        test.issues.append("Process builder: " + pb.toString + NEW_LINE)
+                    }
+                    if (fullPath !== null) {
+                        test.issues.append("File: " + fullPath + NEW_LINE)
+                    }
+                }
+            }
+            case TS: {
+                val dist = test.fileConfig.getSrcGenPath.resolve("dist")
+                val file = dist.resolve(nameOnly + ".js")
+                if (Files.exists(file)) {
+                    pb = new ProcessBuilder("node", file.toString)
+                } else {
+                    test.result = Result.NO_EXEC_FAIL
+                    if (pb !== null) {
+                        test.issues.append("Process builder: " + pb.toString + NEW_LINE)
+                    }
+                    if (file !== null) {
+                        test.issues.append("File: " + file + NEW_LINE)
+                    }
+                }
+            }
+        }
+        if (pb !== null) {
+            try {
+                val p = pb.start()
+                val stdout = test.exec.recordStdOut(p)
+                val stderr = test.exec.recordStdErr(p)
+                if(!p.waitFor(MAX_EXECUTION_TIME_SECONDS, TimeUnit.SECONDS)) {
+                    stdout.interrupt()
+                    stderr.interrupt()
+                    p.destroyForcibly()
+                    test.result = Result.TEST_TIMEOUT
+                } else {
+                    if (p.exitValue == 0) {
+                        test.result = Result.TEST_PASS
+                    } else {
+                        test.result = Result.TEST_FAIL
+                    }
+                }
+                
+            } catch(Exception e) {
+                test.result = Result.TEST_FAIL
```

Comment:
This is actually what happens. We record standard out and standard error in the `LFTest` object (see lines 346-347).

Reply:
Yes, but that only works if the binary actually executed. If there is an exception in the execution (for instance because the binary does not exist), the exception is caught here, but there is no visible output indicating the reason why the test failed. I had this issue with the Windows tests, where it couldn't find the binary (due to missing .exe) and I had no clue why.